### DISCUSSION TITLE
Div Child Not Listening To The Parent Width

### DISCUSSION THREAD

**Discussion by abbagana** [dompdf/dompdf#2829] [General]
I have a main container with four children in it. The main container has the  following properties:
```
    width: 100% !important;
    max-width: 100%!important;
```

while the child has the following
```
    float: left;
    padding: 8px 18px 1px 18px;
    border: 1px solid #111;
    width: 44%;
    height: auto;
    margin-left: 4px;
```

However, the children do not follow the width property of the container the keep on floating without breaking and moving to the next line
![flot image](https://user-images.githubusercontent.com/31752841/159709636-234484ad-4be2-424a-9b42-96e972b81de6.PNG)

**Comment by bsweeney** [comment]
This is an issue with Dompdf not correctly repositioning adjacent floats that reach the edge of the page. It's a known issue though didn't have a dedicated ticket until now (#2831).

If you can use it `display: inline-block` will work for the provided example.

**Comment by bsweeney** [comment]
And now I see you created an issue as well. No need to create both. Discussions are preferred if you're not certain of a bug, we can always create an issue later.
